scaladoc/src/dotty/tools/scaladoc/site/SidebarParser.scala

```diff
@@ -30,16 +32,43 @@ object Sidebar:
 
   private object RawInputTypeRef extends TypeReference[RawInput]
 
-  private def toSidebar(r: RawInput)(using CompilerContext): Sidebar = r match
+  private def pageWithNoTitle(content: String | java.io.File): String =
+    val fileContent = content match {
+    case file: java.io.File => Source.fromFile(file).getLines().mkString("\n")
+    case str: String => str
+    }
+    val lines = fileContent.split("\n")
+    lines.zipWithIndex
+    .find { case (line, i) => line.trim.startsWith("page:") && !lines(i - 1).contains("- title:") }
```

Comment:
I think the validation should be done in two steps: first, we parse the content as YAML, and second, we validate the YAML content. Here, this code is navigating the YAML data based on its text representation, which is very fragile (fore instance, in your code you make assumptions about the order of the fields `page` and `title`). Would it be possible for `pageWithNoTitle` to take as a parameter a YAML data structure?